tests/test_self.py

```diff
@@ -1290,3 +1290,15 @@ def test_regex_paths_csv_validator() -> None:
         with pytest.raises(SystemExit) as ex:
             Run(["--ignore-paths", "test", join(HERE, "regrtest_data", "empty.py")])
         assert ex.value.code == 0
+
+    def test_regression_recursive(self):
+        self._test_output(
+            [join(HERE, "regrtest_data", "directory", "subdirectory"), "--recursive=n"],
+            expected_output="No such file or directory",
+        )
+
+    def test_recursive(self):
+        self._runtest(
+            [join(HERE, "regrtest_data", "directory", "subdirectory"), "--recursive=y"],
```

Comment:
Could you add a test for the `pylint .` behaviour? I think there are some tests already where we change the `curdir` at the start of the test and then move back to the old `curdir` afterwards.

Reply:
done.